### PR TITLE
feat(shared): zod-typed responses for apps lifecycle routes

### PR DESCRIPTION
**Pull request by Dexploarer**
## Summary

Closes the typed-routes loop on the **response side** for the apps-routes surface (request side migrated in #7557 + #7559). Each response is now expressed as a zod schema in \`@elizaos/shared\`, with the inferred type used at the handler return site so future drift is a compile error.

## New response schemas (in \`apps-lifecycle-routes.ts\`)

| Schema | Route | Shape |
|---|---|---|
| \`PostOverlayPresenceResponseSchema\` | POST /api/apps/overlay-presence | \`{ ok: true, appName: string \\| null }\` |
| \`PostCreateAppResponseSchema\` | POST /api/apps/create | \`{ success, text, messages, data }\` |
| \`PostRefreshAppsResponseSchema\` | POST /api/apps/refresh | \`{ ok: true, count }\` |
| \`PostInstallAppResponseSchema\` | POST /api/apps/install | discriminated union on \`success\` |
| \`InstallProgressEventSchema\` | (component) | mirrors \`InstallProgressLike\` |

Plus consume the existing \`FavoritesResponseSchema\` at the response side for GET/PUT \`/api/apps/favorites\` and POST \`/api/apps/favorites/replace\`.

## Out of scope

Routes that delegate to \`AppLaunchResult\` / \`AppStopResult\` (launch, relaunch, stop) — those tree types in \`apps.ts\` are still plain TypeScript interfaces. Migrating them is a separate, deeper pass that should model \`AppViewerConfig\` / \`AppSessionState\` / \`AppRunSummary\` / \`AppLaunchDiagnostic\` together.

## Approach

Compile-time-only typing, no runtime parse-on-write. The handler builds a typed local of \`PostXResponse\`, then passes it to \`json()\`. This catches drift at compile time without risking a 500 from a schema-vs-handler mismatch in production.

## Test plan

- [x] \`bun run test\` in \`@elizaos/shared\` — 11 files, 197 tests pass (26 new)
- [x] \`bun turbo run build --filter=@elizaos/shared\` — clean
- [x] \`bun run typecheck\` in \`@elizaos/agent\` — no new errors in changed files
- [x] Biome lint clean
- [ ] Reviewer: spot-check \`PostInstallAppResponseSchema\` failure branch — \`error\` is optional because the handler permits omission when the installer rejects without an error message

🤖 Generated with [Claude Code](https://claude.com/claude-code)

<details><summary><h3>Greptile Summary</h3></summary>

This PR closes the typed-response loop for the apps-lifecycle routes by adding Zod schemas for five response shapes (`PostOverlayPresenceResponseSchema`, `PostCreateAppResponseSchema`, `PostRefreshAppsResponseSchema`, `InstallProgressEventSchema`, `PostInstallAppResponseSchema`) to `@elizaos/shared`, and adds typed local variables at each handler response site in `apps-routes.ts` so that drift becomes a compile error.

- **New response schemas** added to `packages/shared/src/contracts/apps-lifecycle-routes.ts` covering the install, create, refresh, overlay-presence, and favorites routes; all use `.strict()` and the discriminated union for the install path is correctly modelled.
- **Handler changes** in `packages/agent/src/api/apps-routes.ts` introduce typed intermediary locals (`const response: PostXResponse = {...}`) before each `json(res, response)` call — purely compile-time enforcement with no runtime parse-on-write.
- **26 new tests** in `apps-lifecycle-routes.test.ts` cover acceptance, rejection, strict-mode, and the nullable-error omission case for all new schemas.
</details>

<h3>Confidence Score: 5/5</h3>

Safe to merge — the change is purely additive compile-time typing with no runtime behavior changes.

All handler logic is unchanged; the new typed locals and Zod schemas are compile-time only (no parse-on-write). The discriminated union for the install response, the strict-mode schemas, and the InstallProgressEventSchema all match the actual handler output shapes and the underlying TypeScript interfaces. Tests cover acceptance, rejection, strict-mode, and the optional-error omission case explicitly called out in the PR checklist.

No files require special attention.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| packages/shared/src/contracts/apps-lifecycle-routes.ts | Adds five new response schemas with correct shapes, strict mode, and a discriminated union for the install path; all exported types correctly mirror the handler output. |
| packages/agent/src/api/apps-routes.ts | Typed intermediary locals added at all relevant response sites; imports wired correctly; no logic changes, only compile-time type enforcement. |
| packages/shared/src/contracts/apps-lifecycle-routes.test.ts | 26 new test cases added covering acceptance, rejection, strict-mode enforcement, null/undefined edge cases, and the optional-error omission case on the install failure branch. |

</details>

</details>

<details><summary><h3>Flowchart</h3></summary>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[POST /api/apps/install] --> B{pluginManager.installPlugin}
    B -->|rejects| C[catch: success=false, error=msg]
    B -->|resolves success=false + needs runtime| D[installPluginDirect fallback]
    D --> E{result.success?}
    C --> E
    B -->|resolves| E
    E -->|false| F["PostInstallAppResponse { success: false, error?, progress }"]
    E -->|true| G["PostInstallAppResponse { success: true, pluginName, version, installPath, requiresRestart, progress }"]
    F --> H[json res 422]
    G --> I[json res 200]

    J[GET /PUT /api/apps/favorites] --> K["FavoritesResponse { favoriteApps }"]
    K --> L[json res 200]

    M[POST /api/apps/overlay-presence] --> N["PostOverlayPresenceResponse { ok: true, appName }"]
    N --> O[json res 200]

    P[POST /api/apps/refresh] --> Q["PostRefreshAppsResponse { ok: true, count }"]
    Q --> R[json res 200]

    S[POST /api/apps/create] --> T["PostCreateAppResponse { success, text, messages, data }"]
    T --> U[json res 200]

    style F fill:#f99,stroke:#c33
    style G fill:#9f9,stroke:#3a3
```
</details>

<sub>Reviews (3): Last reviewed commit: ["Merge branch &#39;develop&#39; into wip/zod-type..."](https://github.com/elizaos/eliza/commit/c4c6dc2f14872d6c86781d4029ac09cd61c4936f) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=31525544)</sub>

### CODE REVIEW

File: packages/shared/src/contracts/apps-lifecycle-routes.ts [hunk 1]
```diff
@@ -156,6 +156,79 @@ export const PostOverlayPresenceRequestSchema = z
...
+  })
+  .strict();
+
+/** /create reply from the unified APP action. */
+export const PostCreateAppResponseSchema = z
+  .object({
+    success: z.boolean(),
+    text: z.string(),
+    messages: z.array(z.string()),
+    data: z.unknown(),
```
[the remark this answers is not in the document]
Comment (Dexploarer) [reply packages/shared/src/contracts/apps-lifecycle-routes.ts]: Tested both forms — they're behaviorally identical, including for the rejection case you flagged:

```
z.unknown() with missing data: false                     // already rejected
z.nullable(z.unknown()) with missing data: false         // also rejected
z.unknown() with data:null: true                         // accepted
z.nullable(z.unknown()) with data:null: true             // accepted
z.unknown() with data:undefined: true                    // accepted
```

Inferred types are also identical: `unknown ⊃ null` so `T | null` collapses to `T` when `T = unknown`. The strict-object behavior already requires the field to exist.

If the goal is documentation ("expect null when no data"), I'm happy to add a JSDoc line to the field instead — that signals intent without the no-op schema change. Want me to do that?
